### PR TITLE
feat: Support P-256 Create DID

### LINKED ISSUE #113: Support P-256 Create DID

### PR DESCRIPTION
**Pull request by talwinder50**
1. Enable P-256 did creation
2. Store private key in. Index DB. 
3. Make Create DID Json pretty. 

closes #113

Signed-off-by: talwinder.kaur <talwinder.kaur@securekey.com>

image

image

### CODE REVIEW

File: cmd/user-agent/src/pages/DIDManagement.vue [hunk 1]
```diff
@@ -76,33 +73,83 @@ SPDX-License-Identifier: Apache-2.0
...
-                        "id":"key-recovery",
-                        "type":"JwsVerificationKey2020",
-                        "value":recoveryKeyset.signaturePublicKey,
-                        "encoding":"Jwk",
-                        "keyType":this.selectType,
-                        "recovery":true
-                    }
-                    ]
-                };
+               let createDIDRequest ={}
```
Comment (fqutishat) [inline cmd/user-agent/src/pages/DIDManagement.vue:76]: can change if else to this                                                                                                                                                                     
               let keyValue ={}
                let recoveryKeyValue ={}
                if (this.selectType == "Ed25519"){
                    keyValue=keyset.signaturePublicKey
                    recoveryKeyValue=recoveryKeyset.signaturePublicKey
                }
                const createDIDRequest = {
                    "publicKeys":[{
                        "id":"key-1",
                        "type":"JwsVerificationKey2020",
                        "value":keyValue,
                        "encoding":"Jwk",
                        "keyType":this.selectType,
                        "usage":["ops","general"]
                    }, {
                        "id":"key-recovery",
                        "type":"JwsVerificationKey2020",
                        "value":recoveryKeyValue,
                        "encoding":"Jwk",
                        "keyType":this.selectType,
                        "recovery":true
                    }
                    ]
                };

Comment (talwinder50) [reply cmd/user-agent/src/pages/DIDManagement.vue]: done

File: pkg/controller/command/didclient/command.go [hunk 2]
```diff
@@ -78,9 +88,30 @@ func (c *Command) CreateDID(rw io.Writer, req io.Reader) command.Error {
...
+
 	for _, v := range request.PublicKeys {
-		opts = append(opts, didclient.WithPublicKey(&didclient.PublicKey{ID: v.ID, Type: v.Type, Encoding: v.Encoding,
-			KeyType: v.KeyType, Usage: v.Usage, Recovery: v.Recovery, Value: base58.Decode(v.Value)}))
+		switch v.KeyType {
+		case Ed25519KeyType:
+			opts = append(opts, didclient.WithPublicKey(&didclient.PublicKey{ID: v.ID, Type: v.Type, Encoding: v.Encoding,
+				KeyType: v.KeyType, Usage: v.Usage, Recovery: v.Recovery, Value: base58.Decode(v.Value)}))
+		case P256KeyType:
+			privateKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
```
Comment (fqutishat) [inline pkg/controller/command/didclient/command.go:99]: create method from line 99 to 112

Comment (fqutishat) [reply pkg/controller/command/didclient/command.go:99]: will help your code coverage

Comment (talwinder50) [reply pkg/controller/command/didclient/command.go]: It made only minor difference.

File: pkg/controller/command/didclient/command.go [hunk 3]
```diff
@@ -102,9 +129,29 @@ func (c *Command) CreateDID(rw io.Writer, req io.Reader) command.Error {
...
+		PrivateKey: didPrivateKey ,
+	}, logger)
 
 	logutil.LogDebug(logger, commandName, createDIDCommandMethod, successString)
 
 	return nil
 }
+
+
+func generateECKeyPair() ([]byte, []byte, error){
```
Comment (fqutishat) [inline pkg/controller/command/didclient/command.go:143]: try to do like this https://github.com/trustbloc/trustbloc-did-method/blob/master/pkg/vdri/trustbloc/vdri.go#L39

Comment (fqutishat) [reply pkg/controller/command/didclient/command.go:143]: will help your unit test

Comment (talwinder50) [reply pkg/controller/command/didclient/command.go]: yes that did the trick now project coverage is green above 85, patch is still low though.

**Pull request merged**